### ISSUE TITLE
read_nordic - maximum azimuthal gap

### ISSUE THREAD

**Issue by shicks-seismo** [obspy/obspy#1909]
The current version of read_nordic doesn't appear to read in maximum azimuthal gap and populate originquality, which is written on the second line of the header.

Example:

>  2016  313 0718 21.8 L  -1.997 -14.342 35.7  SOT 16 1.1 6.0LSOT                1
>  GAP=195        1.03      27.7     7.6 33.6 -0.1871E+03 -0.2096E+03  0.8193E+03E
>                                                                                6
>  STAT SP IPHASW D HRMM SECON CODA AMPLIT PERI AZIMU VELO AIN AR TRES W  DIS CAZ7
>  S06D     P   1    718 29.40                             129    0.66 9 37.4 345
>  S06D     S   2    718 34.10                             129    0.06 6 37.4 345

**Comment by calum-chamberlain** [comment]
I can probably have a go at that next week if @shicks-seismo hasn't done it already?  There are quite a few things in the nordic format that I didn't implement, some of them because I didn't find an obvious place for them in events (but that is likely because I didn't look too hard), so if you notice any other things that could be read in (if there) happy to include them too.

**Comment by shicks-seismo** [comment]
Hi Calum,

That would be great if you could. I don’t quite have the time just now, and it would take me a bit of time to delve into the reading of the second line header.

**Comment by calum-chamberlain** [comment]
Currently not included are line types:
- [ ] E (hyp error estimates)
- [ ] F (Fault plane solution)
- [ ] H (High accuracy hypocenter line) - *note, this should probably be preferred_origin if included*
- [ ] M (Moment tensor solution)
- [ ] E13 and EC3 line (explosion info)
- [ ] MACR03 line

I will go ahead and try and implement a load of these where I can find appropriate space, and update the docs to highlight what I don't end up implementing.  The current implementation is a bit bare-bones and should be better.

**Comment by calum-chamberlain** [comment]
On the Error line note, before I write a function to convert from an error ellipse and co-variance in x, y, z to one with semi-major and semi-minor axes lengths and orientation angles, does ObsPy already have one, and if not, where would you like me to put it?

**Comment by megies** [comment]
> one with semi-major and semi-minor axes lengths and orientation angles, does ObsPy already have one, and if not, where would you like me to put it?

Not sure what you mean, the `ConfidenceEllipsoid` should have everything according to QuakeML angle notations: https://github.com/obspy/obspy/blob/master/obspy/core/event/base.py#L1219-L1226

**Comment by calum-chamberlain** [comment]
Sorry, that wasn't clear - Nordic files contain error in x, y, z and covariances. ConfidenceEllipsoid has proper ellipsoid coordinates.  It's simple to convert between the two, I just wondered if there was already an obspy function to convert between the two.  I will probably just write one for now.

**Comment by megies** [comment]
Hmm.. I'm not aware of such a routine, but maybe asking on the users mailing list might be worth a shot.. in any case, I'd also be interested in such a routine, it could e.g. also be used in the NonLinLoc event read routine.

**Comment by calum-chamberlain** [comment]
I'm going to carry this on in #1924 if that is okay with all?

**Comment by megies** [comment]
Sure, fine. For the future, there's no strict need to close the original issue manually, you can put 'closes #1909' in the PR description and once that PR is merged into master, the referenced issue will be automatically closed by github. In any case, it's also fine with me to close this one right away.
